### ISSUE TITLE
CPU Core affinitty

### ISSUE THREAD

**Issue by albvar** [sabnzbd/sabnzbd#1337]
I recently converted to Ubuntu server 19.04 w/ SabNzbd and noticed that the last core on a 6 core VM pegs at 100% and my download speeds (typically 70MB/s+ on a slow day) are being bottlenecked by a single core down to around 10MB/s-30MB/s. I am on a 1Gbps fiber connection from AT&T.
I was running on a windows physical sever prior and was not observing this issue with a consistent 70+MB/s on average.
![image](https://user-images.githubusercontent.com/10334630/67452279-ae7e9c00-f5f0-11e9-899d-233a86e8f250.png)

**Comment by thezoggy** [comment]
check to see what process is pegging your cpu, unrar? par? 
and if you just see high cpu in general during network xfer..
https://kb.vmware.com/s/article/2055140

and side note, I have 1Gig connection and get 100+MB/s in sab and use pretty minimal cpu on an intel i7 4790K with windows

**Comment by Safihre** [comment]
@albvar Can you give an update on this? Did you try @thezoggy's suggestions?
I also have a slow i5 and it get's 100MB/s also.

**Comment by albvar** [comment]
I was using a windows host with sabnzbd prior and I did not notice this issue. I switched over to a ubuntu based VM and a beefier host Dell r720xd running on VMWare workstation pro and that's when the issue became very noticeable. I looked at the CPU utilization for the last core it seemed that python was driving most of the utilization. I tried disabling TSO on the intel adapters and did not see a difference. In all honesty I didn't give it a fair chance as I did not reboot the host, only the guest and issue remained. I eventually ended up switching a a different download client where the core affinity appears to still be present but not to the same degree 90-95% spikes and 70-75MB/s compared to the 10-30MB/s and 100% utilization for a long period of time. This was easy enough for me to implement as docker was facilitation the management of the settings/install etc. From what I have seen, the last core tends to be the core that handles kernel level utilization where disk / network I/O tend to drive usage. I forgot where I encountered this so I am not able to go in a very detailed explanation. I liked sabnzbd'd UI a lot better but C++ VS Python's processing differences allows me to tuck this issue under the rug for now.

**Comment by albvar** [comment]
I was using a windows host with sabnzbd prior and I did not notice this issue. I switched over to a ubuntu based VM and a beefier host Dell r720xd running on VMWare workstation pro and that's when the issue became very noticeable. I looked at the CPU utilization for the last core it seemed that python was driving most of the utilization. I tried disabling TSO on the intel adapters and did not see a difference. In all honesty I didn't give it a fair chance as I did not reboot the host, only the guest and issue remained. I eventually ended up switching a a different download client where the core affinity appears to still be present but not to the same degree 90-95% spikes and 70-75MB/s compared to the 10-30MB/s and 100% utilization for a long period of time. This was easy enough for me to implement as docker was facilitation the management of the settings/install etc. From what I have seen, the last core tends to be the core that handles kernel level utilization where disk / network I/O tend to drive usage. I forgot where I encountered this so I am not able to go in a very detailed explanation. I liked sabnzbd'd UI a lot better but C++ VS Python's processing differences allows me to tuck this issue under the rug for now. The idea to switch download clients came from others (discord) and it ended up being a band-aid I was willing to live with for now.
